Feature/image

RaonJena99 opened the pull request:
1. 유저가 그린 숫자 그림은 프로젝트의 루트에 있는 static/images 폴더에 저장
2. "/images" endpoint를 만들어서 mlflow에서 백엔드에 저장된 유저그림을 zip파일로 받을 수 있음
3. Prometheus를 middleware로 추가해서 로그들을 수집


**수집되는 로그 목록**
- 모델 관련 API 요청 수
- 모델 관련 API 요청 처리 시간(초)
- 모델 관련 API 에러 발생 수

- 캡차 관련 API 요청 수
- 캡차 관련 API 요청 처리 시간(초)
- 캡차 관련 API 에러 발생 수
- 생성된 캡차 총 개수
- 성공적으로 통과한 캡차 개수
- 실패한 캡차 인증 개수
- 유효하지 않은 캡차 ID로 요청한 횟수
- 원격 ML 서비스 호출 지연 시간(초)
- 원격 ML 호출 실패 횟수
- 원격 ML 서비스로 전송된 페이로드 바이트 크기
- API 요청 페이로드 크기(바이트)
- API 응답 바디 크기(바이트)

- 이미지 API(다운로드) 요청 수
- 이미지 API 요청 처리 시간(초)
- 이미지 API 에러 발생 수
- 이미지 다운로드(zip) 요청 수
- 생성된 ZIP 파일 크기(바이트)

RaonJena99 committed:
- download images

- 저장된 유저의 그림을 다운 받을 수 있도록 endpoint 추가

RaonJena99 committed:
-Stash Change Restore

RaonJena99 committed:
이미지 저장 위치 경로 수정

- 이미지를 불러올 경로 수정

RaonJena99 committed:
로그 수집

- 기본 로그 및 필요할 것 같은 로그들 수집

ljy2855 commented on src/routers/captcha.py:
```diff
@@ -1,66 +1,222 @@
             response = await client.post(remote_url, json=payload)
+            ml_duration = time.monotonic() - ml_start
+
+            REMOTE_ML_LATENCY.labels(model="HybridCNN", method=method).observe(
```
모델 predict 실행할때,

https://mlflow.org/docs/latest/deployment/deploy-model-locally/

요기에 있는 방식으로 해주세요! 
아니면 해당 server에서 log된 모델을 불러와서 pyfunc로 실행할수도 있어요. 편한대로 ml쪽 분들이랑 얘기해서 적용해주세요.
https://mlflow.org/docs/latest/api_reference/python_api/mlflow.pyfunc.html

ljy2855 reviewed (commented):
작업 빠르게 해주셔서 감사합니다 :)

ML팀원분들하고 얘기하셔서, mlflow 로컬에서 한번 연동하시고, 연결 한번만 부탁드릴게요!

RaonJena99 committed:
Local Inference Server

- 기존 mlflow에 요청하는 post 명령을 Local Inference Server로 하도록 수정
- ML 서버의 URL은 .env 파일에 저장

RaonJena99 commented:
일단 진용님이 말씀하신 predict 방식은 수정 완료했습니다.
ML 분들께도 관련 내용 전달했습니다.

RaonJena99 commented:
일단 로컬로 프론트,백엔드,ML 서버를 열어서 실행해봤고, 정상적으로 작동되는 건 확인했습니다.

RaonJena99 committed:
"/images" 엔드포인트 수정

- png파일 뿐만 아니라 (png파일과 정답을 맵핑한 csv파일)도 같이 zip으로 전달

RaonJena99 commented:
추가로 ML 쪽의 의견에 따라 "/images" 엔드포인트로 zip파일을 보낼 때, png와 정답을 맵핑한 csv 파일도 같이 전달하도록 수정했습니다.

ljy2855 commented:
수고하셨어요! 추가 작업 따로 없으시면 merge 부탁드려요~
